src/cdk/tree/tree.ts

```diff
@@ -216,12 +350,129 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
       dataStream = observableOf(this._dataSource);
     }
 
-    if (dataStream) {
-      this._dataSubscription = dataStream
-        .pipe(takeUntil(this._onDestroy))
-        .subscribe(data => this.renderNodeChanges(data));
-    } else if (typeof ngDevMode === 'undefined' || ngDevMode) {
-      throw getTreeNoValidDataSourceError();
+    if (!dataStream) {
+      if (typeof ngDevMode === 'undefined' || ngDevMode) {
+        throw getTreeNoValidDataSourceError();
+      }
+      return;
+    }
+
+    let expansionModel;
+    if (!this.treeControl) {
+      this._expansionModel = new SelectionModel<K>(true);
+      expansionModel = this._expansionModel;
+    } else {
+      expansionModel = this.treeControl.expansionModel;
+    }
```

Comment:
Initializing the expansion model seems like it could be its own setup function, rather than being in `_subscribeToDataChanges`